src/Neo/SmartContract/Native/ContractMethodAttribute.cs

```diff
@@ -32,6 +32,11 @@ public ContractMethodAttribute(Hardfork activeIn)
             ActiveIn = activeIn;
         }
 
+        public ContractMethodAttribute(Hardfork activeIn, Hardfork deprecatedIn) : this(activeIn)
```

Comment:
This is optional?

Reply:
You have another method without `deprecatedIn`

Reply:
I think but constructor is unnecessary, if you make `deprecatedIn` `public` property. You can than do 

```csharp
[ContractMethodAttribute(Hardfork.HF_Domovoi, DeprecatedIn = Hardfork.HF_Aspidochelone)]
public void SomeMethod()
{
}
```